sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -731,14 +731,46 @@ def _parse_window_spec(spec):
       return beam.WindowInto(window_fn)
 
   @staticmethod
-  def log_for_testing():
+  def log_for_testing(
+      level: Optional[str] = 'INFO', prefix: Optional[str] = ''):
```

Comment:
`prefix` could default to name of transform (i.e. LogForTesting) or maybe even always put that as part of the log string and append this `prefix` (defaulting to empty string) to that (i.e. `INFO:root:LogForTesting` or `INFO:root:LogForTesting:prefix`) to explicitly show where the log is coming from.

Similar for the Java version, although I'm not sure what is included in the log string by default in that case (perhaps the source class is already included)

Reply:
It's not easy to get at the name of the transform here, runners often prepend the name of the transform in their logs.

Reply:
I just mean putting the literal string 'LogForTestnig' since that is the name of the transform, but not a blocker, just a thought. LGTM for now